File: backend/src/main/java/com/medcollect/api/services/FirebaseAuthService.java
package com.medcollect.api.services;

import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseAuthException;
import com.google.firebase.auth.FirebaseToken;
import com.google.firebase.auth.UserRecord;
import com.medcollect.api.models.User;
import com.medcollect.api.repositories.UserRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.Optional;
import java.util.Set;

@Service
@RequiredArgsConstructor
public class FirebaseAuthService {
    private final FirebaseAuth firebaseAuth;

    public Optional<UserRecord> getUser(String firebaseUid) {
        try {
            return Optional.of(firebaseAuth.getUser(firebaseUid));
        } catch (FirebaseAuthException e) {
            return Optional.empty();
        }
    }
}

File: backend/src/main/kotlin/com/medcollect/api/services/FirebaseAuthService.kt
package com.medcollect.api.services

import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.auth.FirebaseAuthException
import com.google.firebase.auth.UserRecord
import org.springframework.stereotype.Service

@Service
class FirebaseAuthService(
    private val firebaseAuth: FirebaseAuth
) {
    fun getUser(firebaseUid: String): UserRecord? = try {
        firebaseAuth.getUser(firebaseUid)
    } catch (e: FirebaseAuthException) {
        null
    }
}
